lib/wasmex/instance.ex

```diff
@@ -49,6 +49,16 @@ defmodule Wasmex.Instance do
     end
   end
 
+  @spec wasi_from_bytes(binary(), %{optional(binary()) => (... -> any())}, %{optional(:args) => [String.t()], optional(:env) => %{String.t() => String.t()}}) ::
+          {:error, binary()} | {:ok, __MODULE__.t()}
+  def wasi_from_bytes(bytes, imports, wasi) when is_binary(bytes) and is_map(imports) and is_map(wasi) do
+    
```

Comment:
todo: here I need to take apart the `wasi` map:

* args: need to make sure its a list of strings
* env: need to map the `Map` from `name => value` pairs to a list of `name=value` strings and make sure neither `name` nor `value` include an equals sign (`=`)